Tests/Common.ps1

```diff
@@ -33,10 +33,15 @@ function Initialize-CommonTestSetup
     param()
 
     $moduleRootPath = Split-Path -Path $PSScriptRoot -Parent
-    . (Join-Path -Path $moduleRootPath -ChildPath 'Tests\Config\Settings.ps1')
+    $settingsPath = Join-Path -Path $moduleRootPath -ChildPath 'Tests\Config\Settings.ps1'
+    . $settingsPath
     Import-Module -Name (Join-Path -Path $moduleRootPath -ChildPath 'PowerShellForGitHub.psd1') -Force
 
-    if ([string]::IsNullOrEmpty($env:ciAccessToken))
+    $originalSettingsSha256Hash = "944D5BB450AD2C49F77DAE6C472FEF774B108CD2CB6A0DBF45EF4BBB7FE25D56"
```

Comment:
Creative solution!

Unfortunately, it's also pretty delicate, because if that file gets a change of any type at some point (maybe a new setting has to be added), it's going to be hard to remember to update this embedded hash as well.

We need to make maintenance of this project as straightforward as possible.  If we're to proceed with this approach, you should also make an additional update to `.\Tests\Config\Settings.ps1` by adding a comment block at the top (below the copyright header) the reminds _maintainers_ that they need to update the `$originalSettingsSha256Hash` in this file should they ever commit a permanent update to the file.

Reply:
I was thinking the same thing. And, the other day I was reading through `Helpers.ps1` and I saw that you already had a `Get-SHA512Hash` function. I also noticed you has a good amount of static configuration information stored in `GitHubConfiguration.ps1`.

So, I thought I might combine the two and reused `Get-SHA512Hash` to generate the value and `Get-GitHubConfiguration` to store/retrieve the value.

I also remembered that you just reworked the build pipeline and thought it would be an interesting idea to have the hash value calculated during the build process and the `GitHubConfiguration.ps1` file updated with the correct value every time. I added a `run-setTestConfigSettingsHash.yaml` to the builds area and set it as the first step in the `azure-pipelines.ci.yaml` script. <-- I honestly have no idea if I did any of that right.

I also added a comment block to the top of `Settings.ps1`, but I don't know if it should still be there now.